video_xblock/tests/unit/test_backends.py

```diff
@@ -335,3 +355,95 @@ def test_download_video_url_delegates_to_proper_xblock_attribute(self, backend,
+        """
+        Test Vimeo's API client GET method.
+        """
+        # Arrange
+        test_body = {'test': 'body'}
+        requests_get_mock.side_effect = [
+            ResponseStub(status_code=200, body=test_body),
+            ResponseStub(status_code=400)
+        ]
+
+        # Act
+        response = self.vimeo_api_client.get(url='test_url')
+
+        # Assert
+        requests_get_mock.assert_called_with('test_url', headers={'Authorization': 'Bearer test_token'})
+        self.assertEqual(response, test_body)
+        self.assertRaises(vimeo.VimeoApiClientError, self.vimeo_api_client.get, url='test_url')
+
+    def test_api_client_post(self):
+        """
+        Test Vimeo's API client POST method.
+        """
+        self.assertRaises(vimeo.VimeoApiClientError, self.vimeo_api_client.post, "test_url", "test_payload")
+
+    @data([{
+        u'hls_link_expires_time': 1498128010,
+        u'name': u'English_captions_video.vtt',
+        u'language': u'en',
+        u'uri': u'/texttracks/1234567',
+        u'link_expires_time': 1498128010,
+        u'hls_link': u'test_hls_captions_url',
+        u'link': u'test_captions_url',
+        u'active': True,
+        u'type': u'subtitles'
+    }])
+    @patch('video_xblock.backends.vimeo.VimeoPlayer.get_transcript_language_parameters')
+    def test_parse_vimeo_texttracks(self, transcripts_data, get_lang_params_mock):
+        """
+        Test Vimeo's texttracks API parsing utility.
+        """
+        # Arrange
+        get_lang_params_mock.return_value = (u'en', u'English')
+
+        # Act
+        parsed = self.vimeo_player.parse_vimeo_texttracks(transcripts_data)
+
+        # Assert
+        self.assertEqual(parsed, [{
+            'lang': transcripts_data[0][u'language'],
+            'label': u'English',
+            'url': transcripts_data[0][u'link']
+        }])
+        self.assertRaises(vimeo.VimeoApiClientError, self.vimeo_player.parse_vimeo_texttracks, [{}])
+
+    @data({"data": [{"test_key": "test_value"}]})
+    def test_get_default_transcripts(self, test_json_data):
+        """
+        Test Vimeo's default transcripts fetching.
+        """
+
+        with patch.object(self.vimeo_player, 'api_client') as api_client_mock:
+            # Arrange
+            api_client_mock.get.return_value = test_json_data
+            type(api_client_mock).access_token = PropertyMock(return_value="test_token")
+
+            # Act
+            transcripts, _message = self.vimeo_player.get_default_transcripts(video_id="test_video_id")
+
+            # Assert
+            self.assertIsInstance(transcripts, list)
+            api_client_mock.get.assert_called_with('https://api.vimeo.com/videos/test_video_id/texttracks')
+
+            type(api_client_mock).access_token = PropertyMock(return_value=None)  # token-not-provided case
+            self.assertRaises(vimeo.VimeoApiClientError, self.vimeo_player.get_default_transcripts)
```

Comment:
Do we really need to split up testing of single method?  (Double arrangement, double method call).